src/bam/record.rs

```diff
@@ -1970,14 +1970,14 @@ impl CigarStringView {
 
     /// Get number of bases softclipped at the beginning of the alignment.
     pub fn leading_softclips(&self) -> i64 {
-        self.to_owned().take().get_softclips()
+        self.to_owned().take().calc_leading_softclips()
```

Comment:
I don't understand why this would still need to_owned and take.

Reply:
`&self` is a reference of a `CigarStringView but we need a `CigarString`-object/reference when calling `calc_leading_softclips`. But there is no function returning a reference of a `CigarString` which is why I need to own the reference first.